### PR TITLE
feature #2558 get attributes on position

### PR DESCRIPTION
**Pull request by tm-linkwerk**

### CODE REVIEW

**Comment by JohnMcLear** [comment]
Looks good to me from initial looks, haven't tested.  I wonder if it's possible to do test coverage?  Perhaps put tests into a plugin or so?

**Comment by tm-linkwerk** [comment]
Unfortunately I'm not yet familiar with testing in etherpad, though I would really appreciate to have test coverage. Shouldn't it be possible to add tests in etherpad itself? At first glance I didn't find any tests for AttributeManager in https://github.com/ether/etherpad-lite/tree/develop/tests.

**Comment by JohnMcLear** [comment]
I guess what you could do is add some functionality to set the bold button to active based on the caret location (and if the font is bold), that way you would test against that functionality?  That's the only thing I can think to do

**Comment by tm-linkwerk** [comment]
Hmm, I have no idea how to add that functionality to the core.

File: src/static/js/AttributeManager.js [hunk 1]
```diff
@@ -123,6 +123,57 @@ AttributeManager.prototype = _(AttributeManager.prototype).extend({
...
+        return attributes;
+      }
+    }
+    return attributes;
+    
+  },
+  
+  /*
+    Gets all attributes at caret position 
+    if the user selected a range, the start of the selection is taken
```
Comment (JohnMcLear) [inline src/static/js/AttributeManager.js]: Missing docs for what is returned See getAttributesOnPosition

**Comment by JohnMcLear** [comment]
Thanks! :)

**Pull request merged**